container-storage-setup.sh

```diff
@@ -1321,6 +1353,15 @@ setup_storage_compat() {
    Fatal "Storage is already configured with ${current_driver} driver. Can't configure it with ${STORAGE_DRIVER} driver. To override, remove ${_STORAGE_OUT_FILE} and retry."
   fi
 
+  # First expand `auto` if in use
+  local mountroot
+  if test -d ${_DOCKER_METADATA_DIR}; then
```

Comment:
In fact this test probably should be if we are in compatibility mode or not. If we are in compatibility mode, then default to /var/lib/docker otherwise default to /var.

For direct mode, may be we can default to /var for now. That will work for CRI-O as well. And once somebody needs to change the path, we can introduce knob CONTAINER_METADATA_PATH.